src/v/bytes/iobuf.cc

```diff
@@ -200,3 +200,22 @@ bool iobuf::operator==(const iobuf& o) const {
     }
     return true;
 }
+
+bool iobuf::operator==(std::string_view o) const {
+    if (_size != o.size()) {
+        return false;
+    }
+    auto lhs_begin = byte_iterator(cbegin(), cend());
+    auto lhs_end = byte_iterator(cend(), cend());
+    auto c = o.data();
+    while (lhs_begin != lhs_end) {
```

Comment:
Moving noahs suggestion here:
"this can probably be made more efficient if you iterate over io fragments instead of invidual bytes because then you could use std::memcmp over a contiguous region."

Reply:
OHHH smart!!!! and you can even `string_view` compare them. you can get a string_view out of a string_view and get vectorization and stuff. good idea.